uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/staff-leave-report.component.js

```diff
@@ -26,16 +26,15 @@ define([
   HRSettings, sharedSettings) {
     $log.debug('Component: staff-leave-report');
 
-    var requestSort = 'from_date ASC';
     var statusUpdateHandlers = {
       delete: removeLeaveRequestFromItsSection,
       cancel: cancelLeaveRequestStatusHandler
     };
     var vm = this;
 
     vm.absencePeriods = [];
-    vm.absenceTypes = {};
-    vm.absenceTypesFiltered = {};
+    vm.absenceTypes = [];
```

Comment:
I see that you have changed the absence types to an array, which I guess is done to be able to display the absence types in the right order, but is it possible to index them too:

```js
vm.absenceTypesIndex = {};
```

And instead of doing:

```js
var absenceType = _.find(vm.absenceTypes, { id: leaveRequest.type_id });
```
and
```html
ng-init="absenceType = (report.absenceTypes | filter : { 'id' : request.type_id })[0]">
```

Do this:

```js
var absenceType = vm.absenceTypesIndex[request.type_id];
```
and
```html
ng-init="absenceType = report.vm.absenceTypesIndex[request.type_id]">
```

Reply:
This would mean 2 copies of the same data. I was thinking of:

```js
vm.absenceTypesLabelsAndUnits = {
  1: { label: 'foo', unit: 'hours' },
  32: {label: 'bar', unit: 'days'}
}
```

Could that be better?